### PR TITLE
add runner android build ci

### PR DESCRIPTION
**Pull request by cccclai**
As title, add a new CI to cover cross compile android build for llama runner

### CODE REVIEW

File: .ci/scripts/build_llama_android.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+set -ex
+
+install_executorch_and_backend_lib() {
+  echo "Installing libexecutorch.a, libportable_kernels.a, libetdump.a, libbundled_program.a"
+  rm -rf cmake-android-out && mkdir cmake-android-out
+  ANDROID_NDK=/opt/ndk
+  BUCK2=$(which buck2)
+  FLATC=$(which flatc) ANDROID_ABI=arm64-v8a
+  cmake -DBUCK2="$BUCK" -DCMAKE_TOOLCHAIN_FILE="$ANDROID_NDK"/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_PLATFORM=android-23 -DCMAKE_INSTALL_PREFIX=cmake-android-out -DCMAKE_BUILD_TYPE=Release -DEXECUTORCH_BUILD_EXTENSION_MODULE=ON -DEXECUTORCH_BUILD_EXTENSION_DATA_LOADER=ON -DEXECUTORCH_ENABLE_LOGGING=1 -DEXECUTORCH_BUILD_XNNPACK=ON -DPYTHON_EXECUTABLE=python -DEXECUTORCH_BUILD_OPTIMIZED=ON -DXNNPACK_ENABLE_ARM_BF16=OFF -Bcmake-android-out .
```
Comment (digantdesai) [inline .ci/scripts/build_llama_android.sh:16]: nit format one option per line? Just too hard to read / parse - something like this

```suggestion
cmake                                                                         \
    -DBUCK2="$BUCK"                                                           \
    -DCMAKE_TOOLCHAIN_FILE="$ANDROID_NDK"/build/cmake/android.toolchain.cmake \
    -DANDROID_ABI=arm64-v8a                                                   \
    -DANDROID_PLATFORM=android-23                                             \
    -DCMAKE_INSTALL_PREFIX=cmake-android-out                                  \
    -DCMAKE_BUILD_TYPE=Release                                                \
    -DEXECUTORCH_BUILD_EXTENSION_MODULE=ON                                    \
    -DEXECUTORCH_BUILD_EXTENSION_DATA_LOADER=ON                               \
    -DEXECUTORCH_ENABLE_LOGGING=1                                             \
    -DEXECUTORCH_BUILD_XNNPACK=ON                                             \
    -DPYTHON_EXECUTABLE=python                                                \
    -DEXECUTORCH_BUILD_OPTIMIZED=ON                                           \
    -DXNNPACK_ENABLE_ARM_BF16=OFF                                             \
    -Bcmake-android-out .
```

Comment (digantdesai) [reply .ci/scripts/build_llama_android.sh:16]: why do we have to disable BF16? Which version of the NDK we have for the CI job?

File: .ci/scripts/build_llama_android.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  cmake --build cmake-android-out -j16 --target install --config Release
+}
+
+build_llama_runner() {
+    echo "Exporting MobilenetV2"
```
Comment (digantdesai) [inline .ci/scripts/build_llama_android.sh:22]: - Running MV2? :p
- Update BUCK2 path
- why hardcode `-j16`?

Comment (cccclai) [reply .ci/scripts/build_llama_android.sh:22]: was just copying my local command...will updated

**Comment by cccclai** [comment]
see the job https://github.com/pytorch/executorch/actions/runs/8477205000/job/23227800863

File: .ci/scripts/build_llama_android.sh (hunk 1 above)
Comment (cccclai) [reply .ci/scripts/build_llama_android.sh]: r26c, I ran into error with DXNNPACK_ENABLE_ARM_BF16=OFF, maybe can try removing it later

Comment (kimishpatel) [reply .ci/scripts/build_llama_android.sh]: yeah, I didnt have to do this. I would suggest try removing it

**Review by kimishpatel — APPROVED** [review]

**Pull request closed without merging**